dataProcessing.py

```diff
@@ -27,15 +27,21 @@ def parse_file(filename, prob_cutoff, db=''):
 # Read data from hitList structure
 def fill_data_dict(nhits, hitList, db):
     if db in ['pdb', 'pfam', 'yeast']:
-        x1, x2, dx, x1t, x2t, y, pcent, name, detail = fill_data(nhits, hitList, db)
+        x1, x2, dx, x1t, x2t, y, pcent, name, detail, hasLongHits = fill_data(nhits, hitList, db)
         data = dict(x1=x1, x2=x2, xm=[beg+dif/2 for beg,dif in zip(x1,dx)], dx=dx, x1t=x1t, x2t=x2t,
```

Comment:
I wonder if `fill_data` should actually return this dictionary? The number of return arguments is getting rather long :)

Reply:
Perhaps return `data` and `hasLongHits` looking at how data is used later?

Reply:
Or you could actually move the call to `filter_short_hits` inside `fill_data`, and add a comment that this is filtering short hits that occur before the first long hit (since other short hits get filtered in the loop, which confused me for a minute).

Reply:
yeah, it is a bit of a mess... I followed your first 2 suggestions, and added some explanations, let me know if it's good enough.